Formula/octave.rb

```diff
@@ -14,17 +13,18 @@ class Octave < Formula
 
   head do
     url "https://hg.savannah.gnu.org/hgweb/octave", :branch => "default", :using => :hg
-    depends_on "mercurial" => :build
+
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
     depends_on "bison" => :build
     depends_on "icoutils" => :build
     depends_on "librsvg" => :build
     depends_on "sundials"
   end
 
   # Complete list of dependencies at https://wiki.octave.org/Building
-  depends_on "automake" => :build
-  depends_on "autoconf" => :build
   depends_on "gnu-sed" => :build # https://lists.gnu.org/archive/html/octave-maintainers/2016-09/msg00193.html
+  depends_on :java => ["1.6+", :build, :test]
```

Comment:
Unsure about that, because it means that users with non-default homebrew path now require Java to install octave.

Reply:
that seems fine to me. if that bothers them they can use the default prefix

Reply:
If they have admin rights, yes.